src/main/java/org/openmrs/addonindex/service/ElasticSearchIndex.java

```diff
@@ -93,27 +93,39 @@ public void index(AddOnInfoAndVersions infoAndVersions) throws IOException {
 	}
 	
 	@Override
-	public Collection<AddOnInfoSummary> search(AddOnType type, String query) throws IOException {
+	public Collection<AddOnInfoSummary> search(AddOnType type, String query, String tag) throws IOException {
 		
 		BoolQueryBuilder boolQB = QueryBuilders.boolQuery();
 		if (type != null) {
-			//Gets all modules of the type specified by variable type
+			//Exact match on type
 			boolQB.filter(QueryBuilders.matchQuery("type", type));
 		}
+		if (tag != null) {
+			
+		SearchResult result = client.execute(new Search.Builder(new SearchSourceBuilder()
+				.size(SEARCH_SIZE)
+				.query(QueryBuilders.matchQuery("tags", tag)).toString())
+				.addIndex(AddOnInfoAndVersions.ES_INDEX)
+				.build());
+		return result.getHits(AddOnInfoAndVersions.class).stream()
```

Comment:
I didn't really get what you meant here. But if you meant that this method is becoming confusing for a user who calls this function due to the ways of getting results by tag(i.e either itself or as a combination) then that's the reason I had used the getallbyTag method. Another way of removing the confusion would be to add a comment which clearly mentions what's happening. I should've done that already.